### PR TITLE
Make post list shortcode independent of the reSt directive

### PR DESCRIPTION
**Pull request by ralsina**
Moved the actual code into a shortcode plugin, made the reSt directive use the handler.

### CODE REVIEW

**Comment by Kwpolska** [comment]
I think you forgot a file.

**Comment by ralsina** [comment]
Or 2

File: nikola/plugins/shortcode/post_list.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+            'post_list_id': post_list_id,
+            'messages': site.MESSAGES,
+            '_link': site.link,
+        }
+        output = site.template_system.render_template(
+            template, None, template_data)
+        return output, template_deps
+
+# Request file name from shortcode (Issue #2412)
+Plugin.handler.nikola_shortcode_pass_filename = True
```
Comment (Kwpolska) [inline nikola/plugins/shortcode/post_list.py]: No newline at the end of file? What editor is this, notepad.exe?

File: nikola/plugins/shortcode/post_list.py [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY
+# KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+# PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
+# OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+# OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+# SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+"""Post list directive for reStructuredText."""
```
Comment (Kwpolska) [inline nikola/plugins/shortcode/post_list.py]: Not quite.

File: nikola/plugins/shortcode/post_list.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+
+import operator
+import os
+import uuid
+
+import natsort
+from docutils import nodes
```
Comment (Kwpolska) [inline nikola/plugins/shortcode/post_list.py]: Are these docutils imports necessary, and if yes, how can we get rid of them?

**Comment by ralsina** [comment]
@Kwpolska I forgot to run falke8 on it, all those should be fixed now.

File: nikola/plugins/shortcode/post_list.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+from nikola.packages.datecond import date_in_range
+from nikola.plugin_categories import ShortcodePlugin
+
+
+class Plugin(ShortcodePlugin):
+    """Plugin for post-list shortcode."""
+
+    name = "post_list"
+
+    def handler(self, start=None, stop=None, reverse=False, tags=None, require_all_tags=False, categories=None,
```
Comment (Kwpolska) [inline nikola/plugins/shortcode/post_list.py]: @ralsina, perhaps move the long docstring from rest/post_list.py here?

**Pull request merged**